lib/ansible/modules/system/systemd.py

```diff
@@ -291,7 +291,7 @@ def main():
     }
 
     for requires in ('state', 'enabled', 'masked'):
-        if requires is not None and unit is None:
+        if module.params[requires] is not None and unit is None:
```

Comment:
I would even change it to:
~~~ python
if module.params.get(requires) is not None and unit is None:
~~~